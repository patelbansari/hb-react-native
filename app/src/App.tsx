import React from 'react'
import { View, StatusBar } from 'react-native'
import AppNavigation from './navigation/AppNavigation'
import { ApplicationStyles, Colors } from './theme'

function App(): React.ReactElement {
  return (
    <View style={ApplicationStyles.screen.mainContainer}>
      <StatusBar
        backgroundColor={Colors.blueyGreyFour}
        barStyle="dark-content"
      />
      <AppNavigation />
    </View>
  )
}

export default App
