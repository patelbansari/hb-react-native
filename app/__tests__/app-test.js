import React from 'react';
import App from '../src/App';
import { render } from '@testing-library/react-native';

describe("renders correctly ", () => {
  test(("Should display correct title for product screen"), async () => {
    const app = render(<App />);
    const productTitle = app.getByTestId('productTitle');
    expect(productTitle.props.children).toBe('Our Products');
  });
  test(("Snapshot app component"), async () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot();
  });
});