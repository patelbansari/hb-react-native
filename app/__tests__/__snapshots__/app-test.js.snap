// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  Snapshot app component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "zIndex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            onLayout={[Function]}
            pointerEvents="box-none"
            style={null}
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "borderBottomWidth": 0,
                    "elevation": 0,
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": Object {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                      "opacity": 1,
                    }
                  }
                />
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "width": 720,
                      }
                    }
                  >
                    <Image
                      source={Object {}}
                      style={
                        Object {
                          "height": 39.42857142857143,
                          "left": 0,
                          "position": "absolute",
                          "width": 48,
                        }
                      }
                    />
                    <Image
                      resizeMode="contain"
                      source={
                        Object {
                          "uri": "https://cdn1.assets.hollandandbarrett.com/favicon-32x32.png?v=a93b975ef9741c4e5b2fd1c4ad24d1dd%22%20type=%22image/png",
                        }
                      }
                      style={
                        Object {
                          "height": 65.71428571428571,
                          "width": 80,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                      "opacity": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            forwardedRef={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={2}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#ffffff",
                              "flex": 1,
                              "paddingVertical": 24.642857142857142,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontFamily": "Montserrat-Bold",
                                "fontSize": 30,
                                "marginStart": 30,
                                "marginTop": 16.428571428571427,
                              }
                            }
                            testID="productTitle"
                          >
                            Our Products
                          </Text>
                          <View
                            onLayout={[Function]}
                          >
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "marginHorizontal": 40,
                                  "marginTop": 32.857142857142854,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "position": "relative",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderRadius": 22.5,
                                        "height": 394.2857142857143,
                                        "width": 300,
                                      },
                                      Object {
                                        "backgroundColor": "#E1E9EE",
                                        "overflow": "hidden",
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
      </View>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
